src/helpers.js

```diff
@@ -7,32 +7,38 @@ import {isJSON} from "@/utils";
  * @returns {object}
  */
 export function getOptions(context, defaultOptions){
-    const numeric = []; // convert these props to float
+    if(!defaultOptions){
+        defaultOptions = context.options || context.config || {};
+    }
+
+    const numeric = ['autoShow']; // convert these props to float
     const wrapper = context.selectTag;
 
     // options from attribute
     let dataAttribute = wrapper.getAttribute(context.atts.init);
     let options = {};
 
     // data attribute doesn't exist or not JSON format -> string
-    if(!dataAttribute || !isJSON(dataAttribute)){
-        // reassign id
-        const id = dataAttribute || wrapper.id || defaultOptions.id;
-        context.id = id;
-        defaultOptions.id = id;
-
-        return defaultOptions;
-    }
-
-    options = JSON.parse(dataAttribute);
-
-    for(const [key, value] of Object.entries(options)){
-        // convert boolean string to real boolean
-        if(value === "false") options[key] = false;
-        else if(value === "true") options[key] = true;
-        // convert string to float
-        else if(numeric.includes(key) && typeof value === 'string' && value.length > 0) options[key] = parseFloat(value);
-        else options[key] = value;
+    const attributeIsNotJSON = !dataAttribute || !isJSON(dataAttribute);
+
+    // data attribute is not json format or string
+    if(attributeIsNotJSON){
+        options = defaultOptions;
+
+        // data attribute exist => string
+        if(dataAttribute) options.id = dataAttribute;
+        else options.id = '';
```

Comment:
I have added a small condition to remove the `id` from default options (for getting the id with the right priority below purpose).